### PR TITLE
Improve notifications view

### PR DESCRIPTION
**Pull request by Nitwel**
This PR is directly connected to the deprecation warning system that will be introduced in the following PR: #19508

Improves the notifications tab by the following:
- Being able to see the notification message (LOL)
- Not instantly redirecting you when clicking on a notification
- Adding pagination, searching, filtering
- Archive all button
- Cleaning up the code

New notifications tab.
grafik

Gif on how you interact with it.
GIF 23 08 2023 14-28-34

### CODE REVIEW

**Comment by DanielBiegler** [comment]
> Being able to see the notification message (LOL)

lmao

Looks nice! We might wanna look into the console warnings?

> Invalid prop: type check failed for prop "clickable". Expected Boolean, got String with value "
> Hello Admin User,
> Admin User has mentioned you in a comment:
> ... etc ...

**Comment by Nitwel** [comment]
I've now also added an `archive all` button.
grafik

File: app/src/lang/translations/en-US.yaml [hunk 1]
```diff
@@ -347,6 +347,7 @@ clear_items: Clear Items
 reset_to_default: Reset to Default
 undo_changes: Undo Changes
 notifications: Notifications
+mark_all_as_read: Archive All
```
Comment (DanielBiegler) [inline app/src/lang/translations/en-US.yaml:350]: I do think `read` and `archive` are different enough in meaning that we should change the key. Agree?

```suggestion
archive_all: Archive All
```

Maybe? Or something like

```suggestion
archive_all_items: Archive All
```

**Review by DanielBiegler — CHANGES_REQUESTED** [review]

**Review by DanielBiegler — DISMISSED** [review]
Yay, lovely!

File: app/src/views/private/components/notifications-drawer.vue [hunk 2]
```diff
@@ -39,34 +42,78 @@
...
+					@click="toggleNotification(notification.id)"
+				>
+					<div class="header">
+						<v-checkbox
+							:model-value="selection.includes(notification.id)"
+							@update:model-value="toggleSelected(notification.id)"
+						/>
+						<div v-tooltip="notification.subject" class="title">{{ notification.subject }}</div>
+						<div class="spacer" />
+						<div class="time">{{ notification.time }}</div>
```
Comment (azrikahar) [inline app/src/views/private/components/notifications-drawer.vue:66]: Thoughts on using `v-text-overflow` component for the title and the `datetime` display for the time? It's working perfectly as it is now, but just wondering should we reuse components more (very pedantic question!)

**Review by azrikahar — COMMENTED** [review]
"Improve" is definitely an understatement here, love it a lot!

Just a tiny question regarding the `Archive All` action button added/mentioned [over here](https://github.com/directus/directus/pull/19503#issuecomment-1689976899), it feels like a text action button slightly uncommon in terms of action buttons in general that are rounded icons (I could be very wrong here), so just tapping in @rijkvanzanten for some thoughts on it.

**Review by br41nslug — DISMISSED** [review]
Attempted to do some xss but unfortunately it seems to get sanitized properly 😂 Great improvement

File: app/src/views/private/components/notifications-drawer.vue (hunk 2 above)
Comment (paescuj) [reply app/src/views/private/components/notifications-drawer.vue]: Yes, we should 👍 
I've updated it to use `v-text-overflow`. The `datetime` display doesn't come with a tooltip when there's an overflow, though. So I think we should move the logic from `datetime` into a composable which we can then reuse and pass to `v-text-overflow` here.

**Comment by paescuj** [comment]
> "Improve" is definitely an understatement here, love it a lot!

Yeah 🚀 

> Just a tiny question regarding the `Archive All` action button added/mentioned [over here](https://github.com/directus/directus/pull/19503#issuecomment-1689976899), it feels like a text action button slightly uncommon in terms of action buttons in general that are rounded icons (I could be very wrong here), so just tapping in @rijkvanzanten for some thoughts on it.

I agree! Overall, the design feels a bit exotic in comparison to the rest of the app. Let's replace the "Archive All" button, I think the rest is fine for now 👍 Did a quick test and I feel like the header bar from the card layout would be a good match here (may additionally serve for sorting):

**Comment by Nitwel** [comment]
The reason for the text based button was that I didn't find an icon that would explain the difference between the archive and archive all button. And a select all button is not an option because we're using pagination so we don't know what id's there are on the next page. Best solution to me would be to find a proper icon for the archieve all button.

**Comment by paescuj** [comment]
> The reason for the text based button was that I didn't find an icon that would explain the difference between the archive and archive all button. And a select all button is not an option because we're using pagination so we don't know what id's there are on the next page.

I see, but it might be enough to be able to select all items on the current page, like in the other layouts 🤔 

>Best solution to me would be to find a proper icon for the archieve all button. @benhaynes thoughts for an icon?
> But I really like the idea of adding search and filtering to notifications so will add that now too.

Yeah, I suggest waiting for @benhaynes and/or @rijkvanzanten input before proceeding with updates to the UI 👍

**Comment by Nitwel** [comment]
Gave it some finishing touches. You are now able to search and filer. I simplified the archieve all button to an icon that should be fitting. And it now reuses as much code as possible.

grafik
grafik

**Comment by benhaynes** [comment]
That's the right icon! Love it...

**Comment by DanielBiegler** [comment]
We should probably consider merging this into our new pre-release-branch instead of main for the bigger release? Agree?

**Review by paescuj — APPROVED** [review]
Nice work @Nitwel! I like the new UI ❤️

**Comment by paescuj** [comment]
> We should probably consider merging this into our new pre-release-branch instead of main for the bigger release? Agree?

I think this should be fine for `main` now 👍 

The amount of actual changes has been reduced in https://github.com/directus/directus/pull/19503/commits/b451b83c932d27d650a39a50b5237abc507bb42c.

**Review by br41nslug — CHANGES_REQUESTED** [review]
Currently the notifications have the CSS style `user-select: none` applied preventing you from selecting the text in the body of the notification. I think this text should be selectable as it may contain information you'd like to copy-paste.

Also this signout button really needs to go, i clicked it an insane amount of times while testing this PR but that doesnt have to be done in this PR.

https://github.com/directus/directus/assets/9389634/646655cc-309d-4eab-ba6e-30a4cd1983a8

Besides that looking very good 😄

**Review by br41nslug — APPROVED** [review]

**Pull request merged**